components/hal/i2s_hal.c

```diff
@@ -318,8 +318,7 @@ void i2s_hal_tdm_set_tx_slot(i2s_hal_context_t *hal, bool is_slave, const i2s_ha
     uint32_t msk = slot_cfg->tdm.slot_mask;
     /* Get the maximum slot number */
     cnt = 32 - __builtin_clz(msk);
-    /* There should be at least 2 slots in total even for mono mode */
-    cnt = cnt < 2 ? 2 : cnt;
```

Comment:
```c
cnt = (cnt < 2) && (slot_cfg->tdm.ws_width != 1) ? 2 : cnt;
```

Reply:
I'd suggest:
```
cnt = ((cnt < 2) && (slot_cfg->tdm.ws_width != 1)) ? 2 : cnt;
```
as depending on the compiler optimization it may end up assigning zero to `cnt`.

Reply:
Could you help to update the commit?